scripts/getChangelog.py

```diff
@@ -1,90 +1,587 @@
-def print_seen():
-    print("  [already seen] " + pr_details["title"] + " (#" + str(pr_details["number"]) + ")")
-
-print()
-prsSeen = set()
-filename = BASE + " - " + HEAD + " changelog.csv"
-outputFile = open(filename, 'w')
-outputFile.write("Type,Merge date,URL,Title,Author,Type,Functionality group\n")
-commits = requests.get("https://api.github.com/repos/" + REPO + "/compare/" + BASE + "..." + HEAD, headers={'Authorization': 'token ' + TOKEN}).json()
-numCommits = len(commits["commits"])
-for i in range(numCommits):
-    sha = commits["commits"][i]["sha"]
-    commit = commits["commits"][i]
-    print("Commit "+ str(i+1) + " of " + str(numCommits))
-    if "Merge pull request #" in commit["commit"]["message"] or "Merge branch" in commit["commit"]["message"]:
-        print("  [is merge commit]")
-        continue
-    pr_match = re.search(r'\(#(\d{4})\)', commit["commit"]["message"])
-    if pr_match:
-        pr_number = pr_match.group(1)
-        if pr_number in prsSeen:
-            print_seen()
-            continue
-        pr_details = requests.get("https://api.github.com/repos/" + REPO + "/pulls/" + pr_number, headers={'Authorization': 'token ' + TOKEN}).json()
-        outputFile.write("PR," + pr_details["merged_at"] + "," + pr_details["html_url"] + ",\"" + pr_details["title"] + "\"," + pr_details["user"]["login"] + "\n")
-        print("  " + pr_details["title"] + " (#" + str(pr_details["number"]) + ")")
-        prsSeen.add(pr_number)
-        continue
-    time.sleep(1.5) # Avoid rate limit
-    prs = requests.get("https://api.github.com/search/issues?q=repo:" + REPO + "+type:pr+is:merged+" + sha, headers={'Authorization': 'token ' + TOKEN}).json()
-    if len(prs["items"]) == 0:
-        outputFile.write("Commit," + commit["commit"]["committer"]["date"] + "," + commit["html_url"] + ",\"" + commit["commit"]["message"].splitlines()[0] + "\"," + commit["committer"]["login"] + "\n")
-        print("  [orphan] " + commit["commit"]["message"].splitlines()[0])
-        continue
-    pr_details = prs["items"][0]
-    if pr_details["number"] in prsSeen:
-        print_seen()
-        continue
-    outputFile.write("PR," + pr_details["pull_request"]["merged_at"] + "," + pr_details["html_url"] + ",\"" + pr_details["title"] + "\"," + pr_details["user"]["login"] + "\n")
-    print("  " + pr_details["title"] + " (#" + str(pr_details["number"]) + ")")
-    prsSeen.add(pr_details["number"])
-outputFile.close()
+# TODO:
+# Make it possible to run the script through workflow file, completely headless, with only base_ref and head_ref as input
+# Make it possible to run the part of the script that creates a list of PRs that need review replies in a workflow file (which posts a comment for each of the relevant PRs/issues when creating a release)
+# i.e. define two headless states: "get_changelog_CSV" and "get_PRs_that_need_review_replies"
+# However, the former should already be possibe by running the script with only base_ref and head_ref as input
```

Comment:
Please either remove the comment or implement it. I don't really want to commit code that still has todo comments